asmcli/commands/install.sh

```diff
@@ -5,8 +5,201 @@ install_subcommand() {
   # TODO: configure (mesh_ca | citadel | private_ca).
   # TODO: configure (mcp | in-cluster) control plane.
 
-  # TODO: install (mesh_ca | citadel | private_ca).
-  # TODO: install (mcp | in-cluster) control plane.
+  local MANAGED; MANAGED="$(context_get-option "MANAGED")"
+  local DISABLE_CANONICAL_SERVICE; DISABLE_CANONICAL_SERVICE="$(context_get-option "DISABLE_CANONICAL_SERVICE")"
 
-  return
+  if [[ "${MANAGED}" -eq 1 ]]; then
+    start_managed_control_plane
+  else
+    install_in_cluster_control_plane
+  fi
+
+  if [[ "$DISABLE_CANONICAL_SERVICE" -eq 0 ]]; then
+    install_canonical_controller
+  fi
+
+  apply_kube_yamls
+  outro
+}
+
+start_managed_control_plane() {
+  local PROJECT_ID; PROJECT_ID="$(context_get-option "PROJECT_ID")"
+  local CLUSTER_LOCATION; CLUSTER_LOCATION="$(context_get-option "CLUSTER_LOCATION")"
+  local CLUSTER_NAME; CLUSTER_NAME="$(context_get-option "CLUSTER_NAME")"
+
+  local CR_IMAGE_JSON; CR_IMAGE_JSON="";
+  if [[ -n "${_CI_CLOUDRUN_IMAGE_HUB}" ]]; then
+    CR_IMAGE_JSON=$(cat <<EOF
```

Comment:
Can this just be

```
CR_IMAGE_JSON='{"image": "${_CI_CLOUDRUN_IMAGE_HUB}:${_CI_CLOUDRUN_IMAGE_TAG}"}'
```

Reply:
Will the variables be correctly substituted in the single quotes?

Reply:
No of course not. That's silly. This should work though.

```
CR_IMAGE_JSON="{\"image\": \"${_CI_CLOUDRUN_IMAGE_HUB}:${_CI_CLOUDRUN_IMAGE_TAG}\"}"

```

Reply:
Done!